app/helpers/sessions_helper.rb

```diff
@@ -0,0 +1,55 @@
+module SessionsHelper
+  # Logs in the user:
+  def log_in(user)
+    session[:user_id] = user.id
+  end
+
+  def remember(user)
+    user.remember
+    cookies.permanent.signed[:user_id] = user.id
+    cookies.permanent[:remember_token] = user.remember_token
+  end
+
+  def current_user
```

Comment:
Can you figure out a way to test it?

Reply:
Also, this method looks super complex. What do you think about this refactor of `current_user`

```
def current_user
   log_in_from_remeber_me_cookie if !session[:user_id] && cookies.signed[:user_id]
   @current_user ||= User.find_by(id: session[:user_id]) if session[:user_id]
end

def log_in_from_remeber_me_cookie
    remebered_user = User.find_by(id: cookies.signed[:user_id])
    log_in(remebered_user) if remebered_user && remebered_user.check_remember_token?(cookies[:remember_token])
end
```